### PR TITLE
[TASK] #3 - improve javascript / use button tags / display spinner

### PR DESCRIPTION
**Pull request by waldhacker1**
* decide if you want button tags (submit)
* decide if you want to show the spinner
* configurable spinner class attribute
* generic submit element and spinner element identification (for javascript) through data attributes
* little javascript optimization

### CODE REVIEW

File: Resources/Private/Templates/NewsletterSubscription/Form.html [hunk 1]
```diff
@@ -121,7 +121,22 @@
 	</f:if>
 
 	<f:comment>Render button</f:comment>
-	<f:form.submit name="submitType" value="<f:translate key='subscribe' />" class="{settings.form.button.subscribe.class}" />
+	<f:if condition="{settings.form.button.subscribe.useButtonTag}">
+		<f:then>
+			<f:form.button name="submitType" class="{settings.form.button.subscribe.class}" data="{identifier: 'submit'}">
+				<f:translate key='subscribe' />
+				<f:if condition="{settings.form.button.subscribe.enableSpinner}">
+					<span style="display: none;" class="{settings.form.button.subscribe.spinnerClass}" data-identifier="spinner"><!-- --></span>
```
Comment (MattiasNilsson) [inline Resources/Private/Templates/NewsletterSubscription/Form.html]: Remove

File: Resources/Private/Templates/NewsletterSubscription/Form.html [hunk 2]
```diff
@@ -121,7 +121,22 @@
...
+				</f:if>
+			</f:form.button>
+		</f:then>
+		<f:else>
+			<f:form.submit name="submitType" value="<f:translate key='subscribe' />" class="{settings.form.button.subscribe.class}" data="{identifier: 'submit'}" />
+			<f:if condition="{settings.form.button.subscribe.enableSpinner}">
+				<span style="display: none;" class="{settings.form.button.subscribe.spinnerClass}" data-identifier="spinner"><!-- --></span>
```
Comment (MattiasNilsson) [inline Resources/Private/Templates/NewsletterSubscription/Form.html]: Remove

File: Resources/Private/Templates/NewsletterSubscription/Form.html [hunk 3]
```diff
@@ -121,7 +121,22 @@
...
+		<f:then>
+			<f:form.button name="submitType" class="{settings.form.button.subscribe.class}" data="{identifier: 'submit'}">
+				<f:translate key='subscribe' />
+				<f:if condition="{settings.form.button.subscribe.enableSpinner}">
+					<span style="display: none;" class="{settings.form.button.subscribe.spinnerClass}" data-identifier="spinner"><!-- --></span>
+				</f:if>
+			</f:form.button>
+		</f:then>
+		<f:else>
+			<f:form.submit name="submitType" value="<f:translate key='subscribe' />" class="{settings.form.button.subscribe.class}" data="{identifier: 'submit'}" />
```
Comment (MattiasNilsson) [inline Resources/Private/Templates/NewsletterSubscription/Form.html]: Replace the tag usage with inline notation instead {f:translate}

File: Resources/Private/Templates/NewsletterSubscription/Form.html [hunk 4]
```diff
@@ -131,22 +146,37 @@
...
+			<f:then>
+				<f:form.button name="submitType" class="{settings.form.button.unsubscribe.class}" data="{identifier: 'submit'}">
+					<f:translate key='unsubscribe' />
+					<f:if condition="{settings.form.button.unsubscribe.enableSpinner}">
+						<span style="display: none;" class="{settings.form.button.unsubscribe.spinnerClass}" data-identifier="spinner"><!-- --></span>
+					</f:if>
+				</f:form.button>
+			</f:then>
+			<f:else>
+				<f:form.submit name="submitType" value="<f:translate key='unsubscribe' />" class="{settings.form.button.unsubscribe.class}" data="{identifier: 'submit'}" />
```
Comment (MattiasNilsson) [inline Resources/Private/Templates/NewsletterSubscription/Form.html]: Replace the tag usage with inline notation instead {f:translate}

File: Resources/Private/Templates/NewsletterSubscription/Form.html [hunk 5]
```diff
@@ -131,22 +146,37 @@
...
+				<f:if condition="{settings.form.button.unsubscribe.enableSpinner}">
+					<span style="display: none;" class="{settings.form.button.unsubscribe.spinnerClass}" data-identifier="spinner"><!-- --></span>
```
Comment (MattiasNilsson) [inline Resources/Private/Templates/NewsletterSubscription/Form.html]: Remove

**Review by MattiasNilsson — CHANGES_REQUESTED** [review]
Thanks for your PR! There is just some smaller nitpicks otherwise really nice 👍

File: Resources/Private/Templates/NewsletterSubscription/Form.html (hunk 1 above)
Comment (waldhacker1) [reply Resources/Private/Templates/NewsletterSubscription/Form.html]: Not sure what you mean. Should i remove the "< !-- -- >"?

Comment (MattiasNilsson) [reply Resources/Private/Templates/NewsletterSubscription/Form.html]: Yes that is correct. I mean there is no need to have it there.

Comment (waldhacker1) [reply Resources/Private/Templates/NewsletterSubscription/Form.html]: Check :) It's already applied within https://github.com/pixelant/pxa_newsletter_subscription/pull/4/commits/6e8679acd8e4d1bf84510193d1900c883df3a8c6

**Comment by MattiasNilsson** [comment]
Great and thanks for your contribution! 👍

**Pull request merged**